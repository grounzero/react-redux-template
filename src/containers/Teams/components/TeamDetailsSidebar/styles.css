#team-details-sidebar div.ui.vertical.menu {
  width: auto;
  font-size: 1em;
  border-radius: 0; }
  #team-details-sidebar div.ui.vertical.menu a {
    padding: 0.5em; }
  #team-details-sidebar div.ui.vertical.menu .item {
    border-radius: 0; }
  #team-details-sidebar div.ui.vertical.menu .route--active {
    font-weight: bold;
    background: rgba(192, 192, 192, 0.25) !important;
    border-top: rgba(192, 192, 192, 0.75) !important;
    border-bottom: rgba(192, 192, 192, 0.5) !important; }
