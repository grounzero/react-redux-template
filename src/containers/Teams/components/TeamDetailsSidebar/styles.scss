#team-details-sidebar {
    div.ui.vertical.menu {
      width: auto;
      font-size: 1em;
      border-radius: 0;
      a {
        padding: 0.5em;
      }
      .item {
        border-radius: 0;
      }
      .route--active {
        font-weight: bold;
        background: rgba(192, 192, 192, 0.25) !important;
        border-top: rgba(192, 192, 192, 0.75) !important;
        border-bottom: rgba(192, 192, 192, 0.5) !important;
      }
    }
  }