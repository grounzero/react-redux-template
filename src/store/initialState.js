const _initialState = {
  global: {
    userInfo: {
      domain: 'express',
      username: 'dev'
    },
    versionInfo: {
      company: 'Sky Graphics & Data Solutions',
      copyright: 'Copyright Â© 2017',
      date: '01 January 2017',
      description: '',
      product: 'Sports All-Stars Stats',
      title: 'Sports All-Stars Stats',
      version: '0.1.0'
    }
  },
  fixtures: {
  },
  teams: {
    models: [],
    loading: false,
    error: false
  }
};

const initialState = process.env.NODE_ENV === 'production' ? window.__INITIAL_STATE__ : _initialState;

export default initialState;