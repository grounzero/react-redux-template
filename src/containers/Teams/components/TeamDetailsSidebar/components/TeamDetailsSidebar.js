import React from 'react';
import PropTypes from 'prop-types';
import Nav from '../../../../../components/Nav';

import '../styles.css';

const TeamDetailsSidebar = ({loading, teams}) => {
  if (loading) {
    return (
      <div className='ui active inverted dimmer'>
        <div className='ui text loader'>Loading</div>
      </div>
    );
  }
  return (
    <div id='team-details-sidebar'>
      <Nav routes={teams} />
    </div>
  );
};

TeamDetailsSidebar.propTypes = {
  loading: PropTypes.bool.isRequired,
  teams: PropTypes.array.isRequired
};

export default TeamDetailsSidebar;