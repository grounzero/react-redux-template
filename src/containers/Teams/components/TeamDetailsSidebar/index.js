import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import * as thunks from '../../thunks';
import TeamDetailsSidebar from './components/TeamDetailsSidebar';

const TeamDetailsSidebarContainer = ({ teams, loading, actions }) => {
    return (
        <TeamDetailsSidebar teams={teams} loading={loading} actions={actions} />
    );
}

const mapStateToProps = (state) => {
    const { loading, models } = state.teams;
    return {
        loading: loading,
        teams: models.map(team => (
            {
                path: `/teams/${team.id}`,
                exact: true,
                nav: {
                    name: team.name,
                    content: team.name
                }
            }
        ))
    };
}

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(thunks, dispatch)
    };
}

TeamDetailsSidebarContainer.propTypes = {
    loading: PropTypes.bool.isRequired,
    teams: PropTypes.array.isRequired,
    actions: PropTypes.object.isRequired
};

export default connect(mapStateToProps, mapDispatchToProps)(TeamDetailsSidebarContainer);