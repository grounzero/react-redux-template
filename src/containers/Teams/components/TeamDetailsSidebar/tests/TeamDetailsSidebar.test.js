import React from 'react';
import renderer from 'react-test-renderer';

import TeamDetailsSidebar from '../components/TeamDetailsSidebar';
import initialState from '../../../../../store/initialState';

const { teams } = initialState;
const actions = {};

it('TeamDetailsSidebar renders correctly', () => {
    const tree = renderer.create(
        <TeamDetailsSidebar teams={teams.models} loading={teams.loading} actions={actions} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});