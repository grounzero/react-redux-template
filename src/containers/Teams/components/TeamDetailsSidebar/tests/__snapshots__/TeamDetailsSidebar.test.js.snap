// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamDetailsSidebar renders correctly 1`] = `
<div
  id="team-details-sidebar"
>
  <nav>
    <div
      className="ui vertical menu"
    />
  </nav>
</div>
`;
